prime-router/src/main/kotlin/history/azure/DatabaseDeliveryAccess.kt

```diff
@@ -141,4 +154,90 @@ class DatabaseDeliveryAccess(
             query.fetchInto(DeliveryFacility::class.java)
         }
     }
+
+    /**
+     * Get a list of matching delivery facilities for reports delivered to a given [receiver]
+     * @todo further work on https://github.com/CDCgov/prime-reportstream/issues/9272
+     * @param receiver the organization that is being checked.
+     * @param receivingOrgSvc is a specifier for the receiving organization's service.
+     * @param sortDir sort the table by date in ASC or DESC order
+     * @param sortColumns sort the table by a specific set of columns
+     * @param since is the OffsetDateTime minimum date to get results for.
+     * @param until is the OffsetDateTime maximum date to get results for.
+     * @return List of matching delivery facilities
+     */
+    fun fetchBulkDeliveryFacilities(
+        receiver: String,
+        receivingOrgSvc: String?,
+        sortDir: SortDir,
+        sortColumns: List<BulkFacilitySortColumn>,
+        since: OffsetDateTime?,
+        until: OffsetDateTime?,
+    ): List<DeliveryFacility> {
+        val reportIds = db.transactReturning { txn ->
+            var filter = ACTION.ACTION_NAME.eq(TaskAction.send)
+                .and(REPORT_FILE.RECEIVING_ORG.eq(receiver))
+
+            if (receivingOrgSvc != null) {
+                filter = filter.and(REPORT_FILE.RECEIVING_ORG_SVC.eq(receivingOrgSvc))
+            }
+            if (since != null) {
+                filter = filter.and(ACTION.CREATED_AT.ge(since))
+            }
+            if (until != null) {
+                filter = filter.and(ACTION.CREATED_AT.lt(until))
+            }
+
+            val query = DSL.using(txn)
+                // Note the report file and action tables have columns with the same name, so we must specify what we need.
+                .select(REPORT_FILE.REPORT_ID)
+                .from(
+                    ACTION.join(REPORT_FILE).on(
+                        REPORT_FILE.ACTION_ID.eq(ACTION.ACTION_ID)
+                    )
+                )
+                .where(filter)
+                .groupBy(REPORT_FILE.REPORT_ID)
+
+            query.fetchInto(ReportId::class.java)
+        }
+
+        return db.transactReturning { txn ->
+            val sortedColumns = sortColumns.map {
+                val column = when (it) {
+                    /* Decides sort column by enum */
+                    BulkFacilitySortColumn.DATE -> ITEM_LINEAGE.CREATED_AT
+                    BulkFacilitySortColumn.PROVIDER -> COVID_RESULT_METADATA.ORDERING_PROVIDER_NAME
+                    BulkFacilitySortColumn.FACILITY -> COVID_RESULT_METADATA.TESTING_LAB_NAME
+                    BulkFacilitySortColumn.SUBMITTER -> COVID_RESULT_METADATA.SENDER_ID
+                    BulkFacilitySortColumn.REPORT_ID -> COVID_RESULT_METADATA.REPORT_ID
```

Comment:
I'm getting an error when trying to order by `REPORT_ID`
```
[2023-05-11T14:16:01.370Z] Exception: PSQLException: ERROR: for SELECT DISTINCT, ORDER BY expressions must appear in select list
[2023-05-11T14:16:01.370Z]   Position: 760
[2023-05-11T14:16:01.370Z] Stack: java.lang.reflect.InvocationTargetException
[2023-05-11T14:16:01.370Z]      at jdk.internal.reflect.GeneratedMethodAccessor565.invoke(Unknown Source)
[2023-05-11T14:16:01.370Z]      at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2023-05-11T14:16:01.370Z]      at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[2023-05-11T14:16:01.370Z]      at com.microsoft.azure.functions.worker.broker.JavaMethodInvokeInfo.invoke(JavaMethodInvokeInfo.java:22)
[2023-05-11T14:16:01.370Z]      at com.microsoft.azure.functions.worker.broker.EnhancedJavaMethodExecutorImpl.execute(EnhancedJavaMethodExecutorImpl.java:55)
[2023-05-11T14:16:01.370Z]      at com.microsoft.azure.functions.worker.broker.JavaFunctionBroker.invokeMethod(JavaFunctionBroker.java:62)
[2023-05-11T14:16:01.370Z]      at com.microsoft.azure.functions.worker.handler.InvocationRequestHandler.execute(InvocationRequestHandler.java:33)
[2023-05-11T14:16:01.370Z]      at com.microsoft.azure.functions.worker.handler.InvocationRequestHandler.execute(InvocationRequestHandler.java:10)
[2023-05-11T14:16:01.370Z]      at com.microsoft.azure.functions.worker.handler.MessageHandler.handle(MessageHandler.java:45)
[2023-05-11T14:16:01.370Z]      at com.microsoft.azure.functions.worker.JavaWorkerClient$StreamingMessagePeer.lambda$onNext$0(JavaWorkerClient.java:92)
[2023-05-11T14:16:01.370Z]      at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[2023-05-11T14:16:01.370Z]      at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2023-05-11T14:16:01.370Z]      at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2023-05-11T14:16:01.370Z]      at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2023-05-11T14:16:01.370Z]      at java.base/java.lang.Thread.run(Thread.java:829)
[2023-05-11T14:16:01.370Z] Caused by: org.jooq.exception.DataAccessException: SQL [select distinct "public"."item_lineage"."created_at", "public"."covid_result_metadata"."ordering_provider_name", "public"."covid_result_metadata"."testing_lab_name", "public"."covid_result_metadata"."sender_id", item_lineage.child_report_id as "reportId" from "public"."covid_result_metadata" join "public"."item_lineage" on ("public"."covid_result_metadata"."report_id" = "public"."item_lineage"."origin_report_id" and "public"."covid_result_metadata"."report_index" = "public"."item_lineage"."origin_report_index") where "public"."item_lineage"."child_report_id" in (cast(? as uuid), cast(? as uuid), cast(? as uuid), cast(? as uuid), cast(? as uuid), cast(? as uuid), cast(? as uuid), cast(? as uuid), cast(? as uuid), cast(? as uuid)) order by "public"."covid_result_metadata"."report_id" asc]; ERROR: for SELECT DISTINCT, ORDER BY expressions must appear in select list
[2023-05-11T14:16:01.370Z]   Position: 760
[2023-05-11T14:16:01.370Z]      at org.jooq_3.15.4.POSTGRES.debug(Unknown Source)
[2023-05-11T14:16:01.370Z]      at org.jooq.impl.Tools.translate(Tools.java:2988)
[2023-05-11T14:16:01.370Z]      at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:639)
[2023-05-11T14:16:01.370Z]      at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:349)
[2023-05-11T14:16:01.370Z]      at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:295)
[2023-05-11T14:16:01.370Z]      at org.jooq.impl.AbstractResultQuery.fetchLazyNonAutoClosing(AbstractResultQuery.java:316)
[2023-05-11T14:16:01.370Z]      at org.jooq.impl.SelectImpl.fetchLazyNonAutoClosing(SelectImpl.java:2866)
[2023-05-11T14:16:01.370Z]      at org.jooq.impl.ResultQueryTrait.collect(ResultQueryTrait.java:357)
[2023-05-11T14:16:01.370Z]      at org.jooq.impl.ResultQueryTrait.fetchInto(ResultQueryTrait.java:1423)
[2023-05-11T14:16:01.370Z]      at gov.cdc.prime.router.history.azure.DatabaseDeliveryAccess$fetchBulkDeliveryFacilities$1.invoke(DatabaseDeliveryAccess.kt:240)
[2023-05-11T14:16:01.370Z]      at gov.cdc.prime.router.history.azure.DatabaseDeliveryAccess$fetchBulkDeliveryFacilities$1.invoke(DatabaseDeliveryAccess.kt:205)
[2023-05-11T14:16:01.370Z]      at gov.cdc.prime.router.azure.DatabaseAccess.transactReturning$lambda$1(DatabaseAccess.kt:105)
[2023-05-11T14:16:01.370Z]      at org.jooq.impl.DefaultDSLContext.lambda$transactionResult0$3(DefaultDSLContext.java:549)
[2023-05-11T14:16:01.370Z]      at org.jooq.impl.Tools$4$1.block(Tools.java:5252)
[2023-05-11T14:16:01.370Z]      at java.base/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3128)
[2023-05-11T14:16:01.370Z]      at org.jooq.impl.Tools$4.get(Tools.java:5249)
[2023-05-11T14:16:01.370Z]      at org.jooq.impl.DefaultDSLContext.transactionResult0(DefaultDSLContext.java:597)
[2023-05-11T14:16:01.370Z]      at org.jooq.impl.DefaultDSLContext.transactionResult(DefaultDSLContext.java:521)
[2023-05-11T14:16:01.370Z]      at gov.cdc.prime.router.azure.DatabaseAccess.transactReturning(DatabaseAccess.kt:105)
[2023-05-11T14:16:01.370Z]      at gov.cdc.prime.router.history.azure.DatabaseDeliveryAccess.fetchBulkDeliveryFacilities(DatabaseDeliveryAccess.kt:205)
[2023-05-11T14:16:01.370Z]      at gov.cdc.prime.router.history.azure.DeliveryFacade.findBulkDeliveryFacilities(DeliveryFacade.kt:125)
[2023-05-11T14:16:01.370Z]      at gov.cdc.prime.router.history.azure.DeliveryFunction.getBulkDeliveryFacilities(DeliveryFunction.kt:253)
[2023-05-11T14:16:01.370Z]      ... 15 more
[2023-05-11T14:16:01.370Z] Caused by: org.postgresql.util.PSQLException: ERROR: for SELECT DISTINCT, ORDER BY expressions must appear in select list
[2023-05-11T14:16:01.370Z]   Position: 760
[2023-05-11T14:16:01.370Z]      at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2676)
[2023-05-11T14:16:01.370Z]      at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2366)
[2023-05-11T14:16:01.370Z]      at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:356)
[2023-05-11T14:16:01.370Z]      at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:496)
[2023-05-11T14:16:01.370Z]      at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:413)
[2023-05-11T14:16:01.370Z]      at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:190)
[2023-05-11T14:16:01.370Z]      at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:177)
[2023-05-11T14:16:01.370Z]      at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
[2023-05-11T14:16:01.370Z]      at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
[2023-05-11T14:16:01.370Z]      at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:219)
[2023-05-11T14:16:01.370Z]      at org.jooq.impl.Tools.executeStatementAndGetFirstResultSet(Tools.java:4140)
[2023-05-11T14:16:01.370Z]      at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:230)
[2023-05-11T14:16:01.370Z]      at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:335)
[2023-05-11T14:16:01.370Z]      ... 33 more
[2023-05-11T14:16:01.370Z] .
```

Reply:
Solved the issue good catch